### PR TITLE
5718 study list edits

### PR DESCRIPTION
**Pull request by RMcNeely**
Reworked PR for #52 

- Removed most of the thunks in studyActions.js
- Not using chart callbacks to manage StudyLegend actions.
- OverlayMenu is no longer a pure component, it was tied implicitly to studies so now it is no longer meant to be shared.
- No longer using Lodash
- Moved CSS changes into their own SASS file

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
 		"eslint-plugin-promise": "^3.7.0",
 		"eslint-plugin-standard": "^3.1.0",
 		"node-sass": "^4.9.0",
+		"lodash": "^4.17.5",
```
Comment (CodeMan99) [inline package.json]: @RMcNeely I thought you wanted to remove this dependency?

File: src/components/UI/StudyUI.jsx [hunk 2]
```diff
@@ -18,38 +18,52 @@ class StudyUI extends React.Component{
 	constructor(props){
 		super(props);
 	}
+
+	componentDidMount(){
+		this.props.syncStudies.bind(this, this.props)
```
Comment (CodeMan99) [inline src/components/UI/StudyUI.jsx]: What is this line? `Function#bind` returns a new `Function` object. However, it seems like the intention is to modify `this.props.syncStudies` in place, which is not possible.

File: src/components/UI/StudyUI.jsx [hunk 3]
```diff
@@ -18,38 +18,52 @@ class StudyUI extends React.Component{
...
 
-		Object.keys(this.props.studyList).map((key) => {
-			if (this.props.studyList.hasOwnProperty(key)){
-				tempStudies.push(this.props.studyList[key]);
-			}
-		});
+		alphabetized.sort((a, b) => {
+		 	if (a.toLowerCase() > b.toLowerCase()) { return 1; }
+		 	else if (b.toLowerCase() > a.toLowerCase()) { return -1; }
+		 	else { return 0; }
```
Comment (CodeMan99) [inline src/components/UI/StudyUI.jsx]: What the heck? Change this crap please. `a.toLowerCase().localeCompare(b.toLowerCase())` will do the same thing. ([mdn localeCompare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare))

File: src/reducers/studyReducer.js [hunk 4]
```diff
@@ -25,61 +27,75 @@ const initialState = {
...
+			})
+		case Types.REMOVE_STUDY:
+			return Object.assign({}, state, {
+				studyOverlay: {
+					show: false,
+					top: 0,
+					left: 0
+				},
+				// studies: hasStx ? CIQ.clone(action.study.stx.layout.studies,{}) : CIQ.clone(state.studyHelper.stx.layout.studies,{})
+				studies: Object.assign({},action.study.stx.layout.studies)
```
Comment (CodeMan99) [inline src/reducers/studyReducer.js]: How is this removing anything? Is this assignment just relying on the library to do the actual removal?

(seems like this needs a comment, maybe some of the other "types" too)

Comment (RMcNeely) [reply src/reducers/studyReducer.js]: The chartengine is actually removing the studies from its layout, this is just resetting the studies object in redux to match the engines layout.

File: src/components/UI/StudyUI.jsx (hunk 2 above)
Comment (RMcNeely) [reply src/components/UI/StudyUI.jsx]: Looks like an extra line of code that I never removed.

File: src/actions/studyActions.js [hunk 5]
```diff
@@ -110,11 +106,29 @@ export function updateStudy(inputs, outputs, parameters){
...
-		let state = getState()
-		if(state.study.studyHelper !== null) {
-			CIQ.Studies.removeStudy(state.study.studyHelper.stx, state.study.studyHelper.sd);
+		let state = getState();
+		for (var id in state.chart.ciq.layout.studies){
+			let sd = state.chart.ciq.layout.studies[id];
+			if (!sd.customLegend) { CIQ.Studies.removeStudy(state.chart.ciq, sd); }
 		}
-    return dispatch({ type: 'REMOVE_STUDY', study: study })
+		state.chart.ciq.draw();
```
Comment (rbruckheimer) [inline src/actions/studyActions.js]: Do you need this draw() here or is it called for each removed study in the above for loop?

File: src/components/UI/StudyLegend.jsx [hunk 6]
```diff
@@ -24,7 +25,7 @@ class StudyLegend extends React.Component{
 					{studies}
 				</div>
 				<div className='cq-placeholder'>
-					<button className='ciq-btn' onClick={this.props.legendButtonAction}><span className='legend-clear-all'>Clear All</span></button>
+					<button className='ciq-btn' onClick={this.props.legendButtonAction.bind(this, ciq)}><span className='legend-clear-all'>Clear All</span></button>
```
Comment (rbruckheimer) [inline src/components/UI/StudyLegend.jsx]: I've never seen bind used with more than one argument :themoreyouknow:

**Pull request merged**